### PR TITLE
Fix nested container controls scaling that have AutoscaleMode as Inherit

### LINKED ISSUE #5957: Child controls in nested split container scale incorrectly on secondary monitor.
* .NET Core Version:  .NET 6 RC1

* Have you experienced this same bug with .NET Framework?:  Have not tested

**Problem description:**
Child controls in nested split container scale incorrectly on secondary monitor. Look at the following steps.
Put a split container control on the form. Put another inside panel 2 of the first split container, dock=full. Put buttons as depicted below. Form’s AutoScaleMode=DPI. All controls anchor top, left.
These two lines run before Application.Run(New Form2).
`Application.SetDefaultFont(New Font("Segoe UI", 9))`
`Application.SetHighDpiMode(HighDpiMode.PerMonitorV2)`

image
Pic 1. Opened on primary monitor set to 100% DPI. Looks fine.

image
Pic 2. Opened on primary monitor set to 150% DPI. Looks fine.
Opened on secondary monitor set to 150% DPI when primary primary monitor is set to 150% DPI. Looks fine.

image
Pic 3. Opened on secondary monitor set to 150% DPI when primary monitor is set at 100% DPI. Sending WM_DPICHANGED on handle created (workaround #4854). Everything is off.

image
Pic 4. Opened on primary monitor set to 100% DPI then dragged onto secondary monitor at 150% DPI. Scaling of child controls in nested split control is off.

**Expected behavior:**
It should look like Pic 2.

**Minimal repro:**
Put a split container control on the form. Put another inside panel 2 of the first split container, dock=full. Put buttons as depicted above. Form’s AutoScaleMode=DPI. All controls anchor top, left. Have primary monitor set to 100% DPI and secondary monitor set at 150% DPI.

### LINKED ISSUE #6152: PerMonitorV2 DPI scaling not firing ScaleControl method for deeply nested layouts
* .NET Core Version: 6.0 RTM

* Have you experienced this same bug with .NET Framework?: No

**Problem description:**

Deeply nested controls don't get `ScaleControl` called when moving to secondary monitor (150% DPI -> 100% DPI) in PerMonitorV2 mode.

**Expected behavior:**

`ScaleControl` is always called for deeply nested controls when DPI changes.

**Minimal repro:**

To be done.

It looks like the following code may be to blame:
https://github.com/dotnet/winforms/blob/83b3228c2fa5bbde44cb1e8e1b36eb6f328f1289/src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs#L688-L697

When reaching one of the deeply nested container controls it looks up the parent and gets the scale factor from it. However, the parent itself at this point has `AutoScaleMode.Inherit` and hence the values are not set and the reported scaling is `(1.0F, 1.0F)`.  That leads to the following code skipping the scaling:

https://github.com/dotnet/winforms/blob/83b3228c2fa5bbde44cb1e8e1b36eb6f328f1289/src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs#L990-L1000

Values in debugger:

image

### PR DESCRIPTION
**Pull request by dreddy-work**
Fix nested `Container Controls` scaling that have `AutoscaleMode` as `Inherit`.  In this change we are making sure to traverse hierarchy all the way up until we get a container parent whose `AutoScaleMode `is not Inherit and use it s `AutoScaleFactor `to scale children.

Fixes #5957.

fixes #6152 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6130)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs [hunk 1]
```diff
@@ -688,7 +688,12 @@ internal void FocusActiveControlInternal()
         private SizeF GetParentAutoScaleFactor()
         {
             Control parentControl = Parent;
-            while (parentControl is not null and not ContainerControl)
+
+            // Traverse thorugh parent hierarchy untill we get a ContainerControl whose AutoScaleMode is not Inherit.
+            // AutoscaleFactor from this parent is used to scale the child controls within its hierarchy.
+            while (parentControl is not null
+                && (parentControl is not ContainerControl containerControl
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs:695]: Can a control be parented to not a container control?

File: src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs [hunk 2]
```diff
@@ -688,7 +688,12 @@ internal void FocusActiveControlInternal()
         private SizeF GetParentAutoScaleFactor()
         {
             Control parentControl = Parent;
-            while (parentControl is not null and not ContainerControl)
+
+            // Traverse thorugh parent hierarchy untill we get a ContainerControl whose AutoScaleMode is not Inherit.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs:692]: typo: until

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs:692]: typo: through

File: src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/ContainerControl.cs:695]: `AutoscaleFactor `is property on `ContainerControl `.  So, we have to traverse parent chain until we get the right scale factor for its children. Also, `ContainerControl `gets `OnFonChanged `events to scale the children. Non-Container controls that receive `OnFontChanged `does not scale control but just set the `Font`.

**Review by RussKie — APPROVED** [review]

**Comment by dreddy-work** [comment]
CTI cleared testing.

**Pull request merged**